modules/internal/String.chpl

```diff
@@ -36,50 +36,98 @@
-      - ``enum``
+This module supports casts from :record:`string` to numeric types. Such casts
+will convert the string to the numeric type and throw an error if the string
+is invalid. For example:
 
-    To learn more about handling these errors, see the
-    :ref:`Error Handling technical note <readme-errorHandling>`.
+.. code-block:: chapel
 
-  .. note::
+  var number = "a":int;
 
-    As of Chapel 1.18, :record:`string` is a multibyte string whose
-    character set is determined by the user's environment variable
-    settings.  This is normally UTF-8.  ASCII strings are a simple
-    subcase of UTF-8 strings, because every ASCII character is a UTF-8
-    character with the same meaning.
+throws an error when it is executed, but
 
-    For testing, we use the following settings.
+.. code-block:: chapel
 
-    ``export LANG=en_US.UTF-8``
+  var number = "1":int;
 
-    ``export LC_COLLATE=C``
+stores the value ``1`` in ``number``.
 
-    ``unset LC_ALL``
+To learn more about handling these errors, see the
+:ref:`Error Handling technical note <readme-errorHandling>`.
+
+
+Activating Unicode Support
+--------------------------
+
+Chapel strings normally use the UTF-8 encoding. Note that ASCII strings are a
+simple subset of UTF-8 strings, because every ASCII character is a UTF-8
+character with the same meaning.
+
+Certain environment variables may need to be set in order to enable UTF-8
+support. Setting these environment variables may not be necessary at all on
+some systems.
+
+.. note::
+
+  For example, Chapel is currently tested with the following settings:
+
+  ``export LANG=en_US.UTF-8``
+
+  ``export LC_COLLATE=C``
+
+  ``unset LC_ALL``
+
+  LANG sets the default character set.  LC_COLLATE overrides it for
+  sorting, so that we get consistent results.  Anything in LC_ALL
+  would override everything, so we unset it.
+
+.. note::
+
+  Chapel currently relies upon C multibyte character support and may work
+  with other settings of these variables that request non-Unicode multibyte
+  character sets. However such a configuration is not regularly tested and
+  may not work in the future.
+
+Lengths and Offsets in Unicode Strings
+--------------------------------------
+
+For Unicode strings, and in particular UTF-8 strings, there are several possible
+units for offsets or lengths:
+
+ * bytes
+ * code points
+ * graphemes
+
+Most methods on the Chapel string type currently work with byte units by
+default. For example, :proc:`~string.length` returns the length in bytes and
+`int` values passed into :proc:`~string.this` are offsets in byte units.
+
+The string type currently supports code point units as well. In some cases these
+units are indicated by the method name, for example with
+:proc:`~string.ulength`. Other methods, including :proc:`~string.this`, support
+arguments of type :record:`codePointIndex` (instead of `int`) in order to
+request code point units.
+
+.. note::
+
+  Support for graphemes units is not yet implemented.
```

Comment:
Maybe "not implemented at this time."  It isn't clear that we will or should, so saying "not yet" might be misleading.